Mathlib/RingTheory/Localization/FractionRing.lean

```diff
@@ -222,6 +222,23 @@ such that `z = f x * (f y)⁻¹`. -/
 noncomputable def lift (hg : Injective g) : K →+* L :=
   IsLocalization.lift fun y : nonZeroDivisors A => isUnit_map_of_injective hg y
 
+theorem lift_unique
+    {L : Type*} [Field L]
+    {g : A →+* L} (hg : Function.Injective g)
+    {f : K →+* L}
+    (hf1 : ∀ x, f (algebraMap A K x) = g x) : IsFractionRing.lift hg = f :=
+  IsLocalization.lift_unique _ hf1
+
+/-- Another version of unique to give two lift maps should be equal -/
+theorem lift_unique'
+    {L : Type*} [Field L]
+    {g : A →+* L} (hg : Function.Injective g)
+    {f1 f2 : K →+* L}
+    (hf1 : ∀ x, f1 (algebraMap A K x) = g x)
+    (hf2 : ∀ x, f2 (algebraMap A K x) = g x) : f1 = f2 :=
+  Eq.trans (lift_unique hg hf1).symm (lift_unique hg hf2)
```

Comment:
I have not thought of that it dosesn't depend on `g`. Thanks!

Reply:
I think this statement is very similar to that of [IsPurelyInseparable.injective_comp_algebraMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/PurelyInseparable.html#IsPurelyInseparable.injective_comp_algebraMap). Do you think it's a good idea to rephrase this as `Function.Injective fun (f : K →+* L) => f.comp (algebraMap A K)` and rename the result as `IsFractionRing.injective_comp_algebraMap`?

Reply:
> I think this statement is very similar to that of [IsPurelyInseparable.injective_comp_algebraMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/PurelyInseparable.html#IsPurelyInseparable.injective_comp_algebraMap). Do you think it's a good idea to rephrase this as `Function.Injective fun (f : K →+* L) => f.comp (algebraMap A K)` and rename the result as `IsFractionRing.injective_comp_algebraMap`?

It's a good statement but the intuition is slightly different from that of the direct form I think? Maybe I can implement both of them?

Reply:
I think it's the same as your `lift_unique'` besides that the condition is `RingHom.comp f1 (algebraMap A K) = RingHom.comp f2 (algebraMap A K)`. It's the same after `ext`.

Reply:
Of course they are same, but I mean 

> I think it's the same as your `lift_unique'` besides that the condition is `RingHom.comp f1 (algebraMap A K) = RingHom.comp f2 (algebraMap A K)`. It's the same after `ext`.

Of course they are the same. I mean that in some circumstances one may only consider "two homomorphisms are equal" rather than "the derived map is injective" but in other circumstances it is opposite.

Reply:
> in some circumstances one may only consider "two homomorphisms are equal" rather than "the derived map is injective"

I mean what `lift_unique'` can do can also be done by `injective_comp_algebraMap`. For example, `apply lift_unique'` can give `∀ x : A, f1 (algebraMap A K x) = f2 (algebraMap A K x)`, on the other hand, `apply injective_comp_algebraMap; ext` can also give `∀ x : A, f1 (algebraMap A K x) = f2 (algebraMap A K x)`.

Reply:
> > in some circumstances one may only consider "two homomorphisms are equal" rather than "the derived map is injective"
> 
> I mean what `lift_unique'` can do can also be done by `injective_comp_algebraMap`. For example, `apply lift_unique'` can give `∀ x : A, f1 (algebraMap A K x) = f2 (algebraMap A K x)`, on the other hand, `apply injective_comp_algebraMap; ext` can also give `∀ x : A, f1 (algebraMap A K x) = f2 (algebraMap A K x)`.

I know they are the same in Lean. I just means they have different intuitions (in mind) so implement both versions may be better, but of course it's not a big deal(: